posts/kubernetes-tutorial-1-intro.md

```diff
@@ -0,0 +1,69 @@
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem.png)
+
+## Simulating a Disaster
+
+Lets say the IT get alerted about an outage of application Blue. An electrical storm has come by that night and shorted the hardware on the server. During the outage IT determines that they will need to move the application to one of the spare servers. 
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem2.png)
+
+During the outage IT will be scrambling to bring one of the spare servers online. Configure access the shared database, network load balancer and allowing firewall rules for the application to run on the new server. Each of these task can be time consuming if not planned for. 
+
+In our example each application can not recover from a single machine failure. Recovery from said disaster will bring outage times and IT time from investigation to recovery. Naturally we'd ask how can be setup our environment such that our application can recover from this scenario?
+
+## Break Down 
+
+In our example losing one machine means having to manually recover. There are many ways to make your recovery time better in this scenario. For instance more equipment to protect against electrical storm or establish a process to build and test a DR plan in case of such disaster. These type of protections don't change the underlying issue which is single machine recovery. Another method used on-prem is to have one hot machine and a cold machine. The hot machine is used until a issue arises where the cold machine to switched to in case of backup. In this way now you can recover from a single failure but you've only moved the bar one tick and still would not be able to recover from a two machine failure. Ideally we wan't to be able to recovery from N machine failures. 
+
+To achieve this K8s changes the way we deploy applications. Instead of having dedicated hardware and manual deployment processes. Our applications have automated deployments and containerized applications. 
+
+[Kubernetes](https://kubernetes.io/) is one of many systems that enable this functionality by creating a distributed cluster of machines that all work together to host applications deployed. 
+
+## Cluster communication
+
+The way K8s operates is to have a pool of machines ready for any application to be deployed onto them. The machines communicate by having the same Kubelet application installed on all the machines. The Kubelet can manage the node it is installed on by running application containers and also has a communication endpoint for controllers nodes to send operations. 
```

Comment:
Define Kubelet for the layman?

Reply:
Updated paragraph to include some explanation: 

>  The Kubelet has the responsibility of managing the node its installed on and communicating to the controller nodes. It has the system permissions to start and stop containers, mount and unmount volumes, and open and close ports etc. Essentially it is in charge of listening for commands sent from the controller nodes and executing the commands sent. This enables each node in the cluster to be orchestrated as a group of machines, otherwise known as a cluster.